downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,548 @@
+    ];
+
+    // Implement the batching logic.
+    isolated function bookLoaderFunction(readonly & anydata[] ids) returns BookRow[][]|error {
+        final readonly & int[] keys = check ids.ensureType();
+        return keys.'map(isolated function(readonly & int key) returns BookRow[] {
+            return bookTable.'filter(book => book.author == key).toArray();
+        });
+    }
+
+    @graphql:ServiceConfig {
+        contextInit: isolated function(http:RequestContext requestContext, http:Request request) 
+        returns graphql:Context {
+            graphql:Context ctx = new;
+            // Register the dataloader with the context using a unique name.
+            ctx.registerDataLoader("bookLoader", new dataloader:DefaultDataLoader(bookLoaderFunction));
+            return ctx;
+        }
+    }
+    service /graphql on new graphql:Listener(9090) {
+        resource function get authors() returns Author[] {
+            return from AuthorRow authorRow in authorTable
+                select new (authorRow);
+        }
+    }
+
+    public isolated distinct service class Author {
+        private final readonly & AuthorRow author;
+
+        isolated function init(readonly & AuthorRow author) {
+            self.author = author;
+        }
+
+        isolated resource function get name() returns string {
+            return self.author.name;
+        }
+
+        // Define a prefetch function.
+        isolated function preBooks(graphql:Context ctx) {
+            dataloader:DataLoader bookLoader = ctx.getDataLoader("bookLoader");
+            // Add the primary key of the data to be fetched to the dataloader.
+            bookLoader.add(self.author.id);
+        }
+
+        isolated resource function get books(graphql:Context ctx) returns Book[]|error {
+            dataloader:DataLoader bookLoader = ctx.getDataLoader("bookLoader");
+            // Obtain the data from the dataloader using the primary key of the data.
+            BookRow[] bookRows = check bookLoader.get(self.author.id);
+            return from BookRow bookRow in bookRows
+                select {id: bookRow.id, title: bookRow.title};
+        }
+    }
+    ```
+
+### Deprecations
+
+#### `graphql` package
+
+- Deprecated the `executeWithType()` method of the `graphql:Client`.
+
+### Improvements
+
+#### `graphql` package
+
+- Added support for the `@deprecated` directive in output objects defined using record types.
+- Added support for printing the GraphQL URL to `stdout`.
+- Added support for generating a subgraph SDL schema at compile time.
+
+#### `log` package
+
+- Made adjustments to the behavior of the log print functions when an error object is provided for the `error` key. Updated the previous version, which displayed only the error message to now provide more detailed information, including both the error message and the stack trace for the current error as well as any underlying errors that contributed to the current error.
```

Comment:
Second sentence seems not correct with the placement of `now`?

Reply:
It gives the meaning that the previous version is updated to the newer version now.

Reply:
`Updated the previous version, which displayed only the error message to now provide more detailed information..`

Isn't it better to word is as follows

`Updated the previous version, which displayed only the error message to provide more detailed information now...`

having `to now` seems confusing